### PR TITLE
CLN: assorted cleanups

### PR DESCRIPTION
**Pull request by jbrockmendel**
Salvaging what I can from some abandoned branches.

### CODE REVIEW

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -794,6 +794,7 @@ Datetimelike
 - Bug in :class:`DatetimeIndex` addition when adding a non-optimized :class:`DateOffset` incorrectly dropping timezone information (:issue:`30336`)
 - Bug in :meth:`DataFrame.drop` where attempting to drop non-existent values from a DatetimeIndex would yield a confusing error message (:issue:`30399`)
 - Bug in :meth:`DataFrame.append` would remove the timezone-awareness of new data (:issue:`30238`)
+- Bug in :meth:`Series.cummin` and :meth:`Series.cummax` with timezone-aware dtype incorrectly dropping its timezone (:issue:`15553`)
```
Comment (topper-123) [inline doc/source/whatsnew/v1.0.0.rst]: This is #30460, right?

**Review by topper-123 — APPROVED** [review]
LGTM.

File: doc/source/whatsnew/v1.0.0.rst (hunk 1 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v1.0.0.rst]: yes

File: pandas/core/reshape/concat.py [hunk 2]
```diff
@@ -472,9 +472,9 @@ def _get_result_dim(self) -> int:
         else:
             return self.objs[0].ndim
 
-    def _get_new_axes(self):
+    def _get_new_axes(self) -> List[Index]:
         ndim = self._get_result_dim()
-        new_axes = [None] * ndim
+        new_axes: List = [None] * ndim
```
Comment (simonjayhawkins) [inline pandas/core/reshape/concat.py]: can you add type parameters at some point. xref #30539

**Review by simonjayhawkins — APPROVED** [review]
@jbrockmendel lgtm

**Pull request merged**